src/plone/restapi/search/handler.py

```diff
@@ -144,8 +144,6 @@ def filter_query(self, query):
             or query.get("sort_on", "") == "effective"  # compatibility with Volto
         ):
             query["sort_order"] = "reverse"
-        elif "sort_order" in query:
-            del query["sort_order"]
```

Comment:
@mamico `filter_query` it's only called when we have the `use_site_search_settings` parameter. So I think it's trying to ignore `sort_order` and use what's in the search control panel.

Reply:
This is a bit odd: when passed in the query, `sort_on` is preserved, but `sort_order` is removed, and there isn't `sort_order` in the search control panel. 

And, in the Volto search UI (see https://demo.plone.org/search?SearchableText=&sort_on=effective&sort_order=reverse) the label reads "date (newest first)". The corresponding REST call is https://demo.plone.org/++api++/@search?b_size=25&sort_on=effective&sort_order=reverse&use_site_search_settings=1, but the returned results look like they're in the wrong order.